pkg/serializer/func_queue.go

```diff
@@ -3,100 +3,52 @@
 
 package serializer
 
-import (
-	"context"
-	"fmt"
-)
-
-var (
-	// NoRetry always returns false independently of the number of retries.
-	NoRetry = func(int) bool { return false }
-)
-
-// WaitFunc will be invoked each time a queued function has returned an error.
-// nRetries will be set to the number of consecutive execution failures that
-// have occurred so far. The WaitFunc must return true if execution must be
-// retried or false if the function must be returned from the queue.
-type WaitFunc func(nRetries int) bool
-
-type queuedFunction struct {
-	f        func() error
-	waitFunc WaitFunc
-}
-
 type FunctionQueue struct {
-	queue  chan queuedFunction
+	queue  chan func() error
 	stopCh chan struct{}
+	err    error
 }
 
 // NewFunctionQueue returns a FunctionQueue that will be used to execute
 // functions in the same order they are enqueued.
 func NewFunctionQueue(queueSize uint) *FunctionQueue {
 	fq := &FunctionQueue{
-		queue:  make(chan queuedFunction, queueSize),
+		queue:  make(chan func() error),
```

Comment:
If this commit is going to ignore the `queueSize` parameter, then it should also remove the parameter so that callers do not expect some different behaviour compared to what the API provides.

Reply:
Thanks, good catch!